drafts/2020-02-20-Rest-client-generator.adoc

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "The OpenAPI Generator has added support for Eclipse MicroProfile"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/u/28316667
+author_github: https://github.com/turkeylurkey
+seo-title: OpenAPI Generator has added support for Eclipse MicroProfile - OpenLiberty.io
+seo-description: Develop microservices quicker using the OpenAPI Generator and the Eclipse MicroProfile programming framework.
+blog_description: "Develop microservices quicker using the OpenAPI Generator and the Eclipse MicroProfile programming framework."
+---
+= The OpenAPI Generator has added support for Eclipse MicroProfile
+Paul Gooderham <https://github.com/turkeylurkey>
+
+The OpenAPI Generator can help a developer access a new REST endpoint from their application. It does this by actually generating part of the code required for the application to access that endpoint. The generator can create Java files and also the directory structure corresponding to the package. Recently the OpenAPI Generator was upgraded to support the Microprofile annotations and is an ideal choice for the Open Liberty developer.
+
+If you want to learn more about Microprofile and Open Liberty please check out the educational guide that’s available. You can learn by working on a sample project and you can complete it in less than an hour. https://openliberty.io/guides/microprofile-rest-client.html[Consuming RESTful Java microservices] Pay attention to the sections "Injecting the client with dependency injection" and "Building the client with RestClientBuilder."
+
+== Example
+Say you own a Java web application that you implemented with Microprofile and it is the front end for a collection of stores. Today you are going to add a pet store. You found a pet store back-end service that you're going to use because it provides a RESTful interface. You can use the OpenAPI Generator to accelerate the integration.
+
+The pet store service documents the API using the OpenAPI specification because it is both human readable and machine readable. You download the API documentation in a file called petstore.yaml. The challenge now is to design and code the Java module that will communicate with the pet store service.
```

Comment:
"You download..." - what's that mean? Does it mean the developer would literally download a copy of the yaml file to their laptop? Where from (how would they know where to find that file to download it?)?

Reply:
Yes, I mean they literally download it. I am presuming they are purchasing a service and the vendor provides some support. For example, they would get on the phone with a sales person, decide on a price, and the vendor would give all the technical details needed. However, a correctly configured service could also make the file available for viewing in a browser and hence you could save it to your laptop. I'm not sure how much real life detail to provide.

Reply:
I think then just don't say "download" because it implies you're giving an instruction to how to obtain the documentation. Just talk about the documentation that's been generated from the API using OpenAPI and that the developer writes the client code from it or, in this case, OpenAPI Generator can generate from it.

Reply:
Made it more generic to say: "the developer provided the API documentation in a file called petstore.yaml."